### PR TITLE
feat: allow configuration of explorer links

### PR DESCRIPTION
**Pull request by DNR500**
JIra: [LF-9882](https://lifi.atlassian.net/browse/LF-9882)

Allows configuration of explorer urls in the Widget

[LF-9882]: https://lifi.atlassian.net/browse/LF-9882?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/widget/src/hooks/useExplorer.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  const getAddressLinkByChainId = (address: string, chainId: number) => {
+    return getAddressLink(address, getChainById(chainId));
+  };
+
+  return {
+    getTransactionLink,
+    getTransactionLinkByChainId,
+    getAddressLink,
+    getAddressLinkByChainId,
```
Comment (DNR500) [inline packages/widget/src/hooks/useExplorer.ts:53]: In some places we already have access to the chain object and in other places with only have access to the chain id so have some additional functions to deal with this

File: packages/widget/src/hooks/useExplorer.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    return `${getBaseUrl(chain)}/tx/${txHash}`;
+  };
+
+  const getTransactionLinkByChainId = (txHash: string, chainId: number) => {
+    return getTransactionLink(txHash, getChainById(chainId));
+  };
+
+  const getAddressLink = (address: string, chain?: Chain) => {
+    // TODO: consider here that LiFi explorer supports /wallet/ rather than /address/
```
Comment (DNR500) [inline packages/widget/src/hooks/useExplorer.ts:36]: Lifi Explorer (and Jumper Scan too) `/wallet/` gets used in place of  `/address/`

https://github.com/lifinance/lifi-explorer/blob/develop/packages/explorer/src/config/routes.ts

In the current WIP version I have set this up so you can see the problem.

1. Connect your wallet
2. Select Arbitrum as your chain in the wallet
3. And then click the explorer link from the widget wallet menu

Should we change the widget configuration on this to allow the tx and address paths to be overridable?

something like this?

```
explorerUrls: {
    42161: {
      baseUrl: 'https://jumper.exchange/scan',
    },
    internal: {
      baseUrl: 'https://jumper.exchange/scan',
      addressPath: '/wallet/',
      txPath: '/transaction/',
   },
}
```

File: packages/widget/src/types/widget.ts [hunk 3]
```diff
@@ -225,6 +225,7 @@ export interface WidgetConfig {
   languages?: WidgetLanguages;
   languageResources?: LanguageResources;
   explorerUrl?: string;
```
Comment (chybisov) [inline packages/widget/src/types/widget.ts:227]: Let's remove this :)

File: packages/widget/src/hooks/useExplorer.ts (hunk 1 above)
Comment (DNR500) [reply packages/widget/src/hooks/useExplorer.ts]: remove the ..ByChainId functions and consolidated that functionality into the other functions

File: packages/widget/src/hooks/useExplorer.ts (hunk 2 above)
Comment (DNR500) [reply packages/widget/src/hooks/useExplorer.ts]: This is a possible option in future should we need it. Staying with the current structure for now

**Review by chybisov — APPROVED** [review]

**Pull request merged**